src/main/java/io/cryostat/platform/ServiceRef.java

```diff
@@ -218,4 +220,52 @@ public enum AnnotationKey {
         REALM,
         ;
     }
+
+    public static Set<ServiceRef> getAddedOrUpdatedRefs(
```

Comment:
What do you think of an interface like this?

```java
Set<ServiceRef> added = ServiceRef.compare(previous).to(curremt).added();
Set<ServiceRef> changed = ServiceRef.compare(previous).to(current).changed();
Set<ServiceRef> removed = ServiceRef.compare(previous).to(current).removed();
```

It's basically just a builder pattern. The implementation would be something like:

```
class ServiceRef {
  ServiceRef.Compare compare(Collection previous) {
    return new Compare(previous);
  }

  static class Compare {
    private Collection previous, current;
    private Compare(Collection previous) {
      this.previous = previous;
    }
    public to(Collection current) {
      this.current = current;
    }

    public added() {
        Set result = new HashSet(current);
        result.removeAll(previous);
        return result;
    }

    ...
  }
}
```